### PR TITLE
Teach the TimerPicker how to pick a time

### PR DESCRIPTION
**Pull request by abarth**
The TimePicker can now actually pick a time. However, it doesn't understand AM
or PM and there's lots of visual polish missing.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

**Comment by abarth** [comment]
I suspect Travis us mad because of the new point release of `package:test`.  We need to pin `test` tighter given that we use some non-public APIs.

**Comment by abarth** [comment]
Test package fix is in https://github.com/flutter/flutter/pull/570

File: packages/flutter/lib/src/material/time_picker.dart [hunk 1]
```diff
@@ -123,3 +148,192 @@ class _TimePickerHeader extends StatelessComponent {
...
+  TextStyle style = Typography.black.subhead.copyWith(height: 1.0);
+  List<TextPainter> painters = new List<TextPainter>(labels.length);
+  for (int i = 0; i < painters.length; ++i) {
+    String label = labels[i];
+    TextPainter painter = new TextPainter(
+      new StyledTextSpan(style, [
+        new PlainTextSpan(label)
+      ])
+    )
+      ..maxWidth = double.INFINITY
```
Comment (Hixie) [inline packages/flutter/lib/src/material/time_picker.dart]: move painter.. up to here, and assign as just one statement.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**